sound/soc/sof/imx/imx8.c

```diff
@@ -332,6 +332,11 @@ static int imx8_remove(struct snd_sof_dev *sdev)
 	return 0;
 }
 
+static int imx8_machine_register(struct snd_sof_dev *sdev)
+{
+	return devm_of_platform_populate(sdev->dev);
```

Comment:
what is this supposed to do?

Reply:
The SOF DSP DT will look like this:

```

dsp {
       compatible = "fsl,imx8qxp-dsp";
       reg = <0x596e8000 0x88000>;
       clocks = <..>
       mach_drv: mach_drv {
                compatible = "simple-scu-audio-card";
                simple-audio-card,widgets =<..>
                simple-audio-card,routing =<..>
                simple-audio-card,dai-link {
                        format = "i2s";
                        codec {
                                sound-dai = <&cs42888>;
                       };
                };
        };
```

devm_of_platform_populate - walks the device tree and creates devices from nodes.

Now, it starts from DSP root node and only mach_drv child node. It reads the node and create a platform_device from it. It is similar with ``platform_device_register_data`` but it reads the information from DT instead of having them passed via pdata argument of platform_device_register_data.